integration/helm_test.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/GoogleContainerTools/skaffold/integration/skaffold"
+	"github.com/google/go-cmp/cmp"
+)
+
+const (
+	TestVersion = "vtest"
+)
+
+func TestHelmDeploy(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+	if os.Getenv("REMOTE_INTEGRATION") != "true" {
+		t.Skip("skipping remote only test")
+	}
+
+	helmDir := "examples/helm-deployment"
+
+	ns, client, deleteNs := SetupNamespace(t)
+	// To fix #1823, we make use of env variable templating for release name
+	env := []string{fmt.Sprintf("TEST_NS=%s", ns.Name)}
+	depName := fmt.Sprintf("skaffold-helm-%s", ns.Name)
+	fmt.Println(env)
+	defer func() {
+		skaffold.Delete().InDir(helmDir).InNs(ns.Name).WithEnv(env).RunOrFail(t)
+		deleteNs()
+	}()
+
+	skaffold.Deploy("--images", "gcr.io/k8s-skaffold/skaffold-helm").InDir(helmDir).InNs(ns.Name).WithEnv(env).RunOrFailOutput(t)
+
+	client.WaitForDeploymentsToStabilize(depName)
+
+	expectedLabels := map[string]string{
+		"app.kubernetes.io/managed-by": TestVersion,
```

Comment:
can you refer to these by their constants? `runner.K8ManagedByLabel` and `constants.Labels.Deployer`

Reply:
using the same constants as in your source code does not test for correctness.
Test code should not depend on development code.
say, you change the constant to something else or introduce a spelling mistake, then the test will always succeed

Reply:
I'm not sure I agree here. I feel like the more important part of this test is testing the label value, not key. I wouldn't really consider this a dependency on production code. but I guess I don't feel strongly enough about it to push you to change it.

Reply:
thanks! understood.
But in this case, i do want to make sure the test has the label  "app.kubernetes.io/managed-by".
This label is used for tracking the metrics.
if we introduce a change to change this label key, it will fail this test which is something i want to.